controllers/ibmvpcmachine_controller.go

```diff
@@ -181,8 +181,11 @@ func (r *IBMVPCMachineReconciler) reconcileNormal(machineScope *scope.MachineSco
 					Address: *floatingIP.Address,
 				})
 			} else {
+				if instance.PrimaryNetworkInterface.PrimaryIP.Address == nil || *instance.PrimaryNetworkInterface.PrimaryIP.Address == "0.0.0.0" {
+					return ctrl.Result{}, errors.Wrapf(err, "invalid primary ip address")
+				}
 				internalIP := instance.PrimaryNetworkInterface.PrimaryIP.Address
-				for _, port := range []int64{6443, 22} {
+				for _, port := range []int64{6443} {
```

Comment:
do we really need loop if only one port?

Reply:
Not needed, thought will be easier to incorporate list of new ports easily in future with this version of the code. will fix it.

Reply:
Fixed.